gravity-forms/gw-zip-files.php

```diff
@@ -90,7 +90,11 @@ public function archive_files( $entry, $form ) {
 			return;
 		}
 
-		$archive_file_paths = wp_list_pluck( $archive_files, 'path' );
+		$archive_file_paths = array();
+		foreach ( $archive_files as $archive_file ) {
+			$archive_file_path  = array_values( wp_list_pluck( $archive_file, 'path' ) );
+			$archive_file_paths = array_merge( $archive_file_paths, $archive_file_path );
+		}
```

Comment:
Does this still work with non-GPNF fields?

Reply:
@claygriffiths Thanks for that. I updated L87 to load `$archive_files = array( $this->get_entry_files( $entry, $form ) );` in the same "array way". Then the merge goes smoothly. So basically I can confirm I have tested:

1. All nested form entries combined zip.
2. Files from nested form entries and parent entry in a combined zip.
3. No nested form entry, only parent form entry zip.

Anything I am missing still?

Reply:
Nice work! This makes sense.

It does feel a bit weird that any calls to `get_entry_files` get wrapped in another array so I'd definitely recommend adding some inline comments to this. Pretty much everything you mentioned in these comments was super helpful so if you could distill it into inline/block comments that'd be perfect.